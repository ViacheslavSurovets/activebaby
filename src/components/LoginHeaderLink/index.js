export { default as LoginHeaderLink } from './LoginHeaderLink';
