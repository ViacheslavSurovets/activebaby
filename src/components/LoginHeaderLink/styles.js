import styled from 'styled-components';
import { desktop } from '@utils/media';
import { styleHeaderItem } from '@components/Header/styles';
import { Link } from 'react-router-dom';

export const WrapperLink = styled ( Link )`
  display: none;
  ${ desktop }{
  ${ styleHeaderItem };
  color: ${ ( { theme } ) => theme.white };
  display: inherit;
  justify-content: space-between;
  text-decoration: none;
  width: 8rem;
  margin-right: 2rem;
  &:hover {
    svg path {
      fill: ${ ( { theme } ) => theme.header.colorOnHover }
    }
  }
  } 
`;

export const LoginText = styled.span` 
`;
