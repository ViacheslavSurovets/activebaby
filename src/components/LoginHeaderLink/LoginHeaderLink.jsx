import React from 'react';
import { LoginIcon } from '@components';
import { LoginText, WrapperLink } from './styles';
import { useTranslation } from 'react-i18next';


const LoginHeaderLink = () => {
  const { t, i18n } = useTranslation ();
  return (
    <WrapperLink to='/auth'>
      <LoginIcon />
      <LoginText>{t('header.links.login')}</LoginText>
    </WrapperLink>
  );
};

export default LoginHeaderLink;
