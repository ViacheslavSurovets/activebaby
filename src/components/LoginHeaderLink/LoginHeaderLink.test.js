import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import LoginHeaderLink from './LoginHeaderLink';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('LoginHeaderLink', ()=> {
  it('LoginHeaderLink Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <LoginHeaderLink  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
