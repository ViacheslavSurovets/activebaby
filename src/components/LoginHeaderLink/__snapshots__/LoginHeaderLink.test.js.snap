// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginHeaderLink LoginHeaderLink Snapshot 1`] = `ShallowWrapper {}`;
